internal/event/machineevent/machineevent.go

```diff
@@ -0,0 +1,154 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package machineevent
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/gogo/protobuf/proto"
+	irievent "github.com/ironcore-dev/ironcore/iri/apis/event/v1alpha1"
+	irimeta "github.com/ironcore-dev/ironcore/iri/apis/meta/v1alpha1"
+	"github.com/ironcore-dev/libvirt-provider/api"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+// Interface for recording events
+type EventRecorder interface {
+	Eventf(log logr.Logger, apiMetadata api.Metadata, eventType string, reason string, messageFormat string, args ...any)
+}
+
+// Interface for listing events
+type EventStore interface {
+	ListEvents() []*irievent.Event
+}
+
+// Options to initialize the machine event store
+type EventStoreOptions struct {
+	MachineEventMaxEvents      int
+	MachineEventTTL            time.Duration
+	MachineEventResyncInterval time.Duration
+}
+
+// eventStore implements the EventRecorder interface and represents an in-memory event store with TTL for events.
+type eventStore struct {
+	maxEvents           int               // Maximum number of events in the store
+	events              []*irievent.Event // Slice of events
+	mutex               sync.Mutex        // Mutex for thread safety
+	eventTTL            time.Duration     // TTL for events
+	eventResyncInterval time.Duration     // Resync interval for event store's TTL expiration check
+	head                int               // Index of the oldest event
+	count               int               // Current number of events in the store
+	log                 logr.Logger       // Logger for logging overridden events
+}
+
+// NewEventStore creates a new EventStore with a fixed number of events and set TTL for events.
+func NewEventStore(log logr.Logger, opts EventStoreOptions) *eventStore {
```

Comment:
Exported function with the unexported return type. Maybe change `eventStore` to `Store`. 
I think together with the package import it's still clear what we are using (`machineevent.Store`)